### PR TITLE
[Merged by Bors] - use zap in mesh package

### PR DESCRIPTION
**Pull request by poszu**
## Motivation

Part of the effort to migrate from log to zap.

## Test Plan

n/a

## TODO

- [x] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Review by ivan4th — APPROVED** [review]

**Review by fasmat — APPROVED** [review]

**Comment by poszu** [comment]
Bors merge

**Comment by poszu** [comment]
Bors merge

**Comment by poszu** [comment]
boot-0 got restarted during the TestSmeshing test.

bors merge

**Comment by poszu** [comment]
Bors merge

**Comment by poszu** [comment]
bors merge

File: mesh/mesh.go [hunk 1]
```diff
@@ -118,25 +119,24 @@ func (msh *Mesh) recoverFromDB(latest types.LayerID) {
...
-		msh.logger.With().Fatal("failed to recover latest applied layer", log.Err(err))
+		msh.logger.Fatal("failed to recover latest applied layer", zap.Error(err))
 	}
 	msh.setLatestLayerInState(applied)
 
 	if applied.After(types.GetEffectiveGenesis()) {
 		if err = msh.executor.Revert(context.Background(), applied); err != nil {
-			msh.logger.With().
-				Fatal("failed to load state for layer", msh.LatestLayerInState(), log.Err(err))
+			msh.logger.Fatal("failed to load state for layer", msh.LatestLayerInState().Field().Zap(), zap.Error(err))
```
Comment (fasmat) [inline mesh/mesh.go]: NIT: either the `Field` method on `LayerID` needs to be updated to return a `zap.Field` or we should do something like this instead:

```suggestion
			msh.logger.Fatal("failed to load state for layer",
				zap.Uint32("layer_id", msh.LatestLayerInState().Uint32()),
				zap.Error(err),
			)
```

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**